packages/flutter/test/cupertino/slider_test.dart

```diff
@@ -253,6 +254,92 @@ void main() {
     expect(SchedulerBinding.instance.transientCallbackCount, equals(0));
   });
 
+  testWidgets(
+    'Slider emits haptic feedback when hitting edge',
+    (WidgetTester tester) async {
+      final List<MethodCall> hapticLog = <MethodCall>[];
+      tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, (
+        MethodCall methodCall,
+      ) async {
+        hapticLog.add(methodCall);
+        return null;
+      });
+
+      final Key sliderKey = UniqueKey();
+      double value = 0.0;
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) {
+              return Material(
+                child: Center(
+                  child: CupertinoSlider(
+                    key: sliderKey,
+                    value: value,
+                    onChanged: (double newValue) {
+                      setState(() {
+                        value = newValue;
+                      });
+                    },
+                  ),
+                ),
+              );
+            },
+          ),
+        ),
+      );
+
+      // No haptic feedback should be emitted when the slider is created.
+      expect(hapticLog, hasLength(0));
+
+      const double unit = CupertinoThumbPainter.radius;
+      final Offset topLeft = tester.getTopLeft(find.byKey(sliderKey));
+      Offset thumbCenter = topLeft + const Offset(unit, unit);
+      const Offset delta = Offset(50.0, 0.0);
+      await tester.dragFrom(thumbCenter, delta);
+      await tester.pump();
+
+      thumbCenter += delta;
+
+      // No haptic feedback should be emitted when the slider is moved.
+      expect(hapticLog, hasLength(0));
+
+      // Move the slider to the end quickly.
+      await tester.timedDragFrom(
+        thumbCenter,
+        const Offset(1000.0, 0.0),
+        const Duration(milliseconds: 100),
+      );
+
+      // Medium haptic feedback should be emitted when the slider is quickly moved to the end.
+      expect(hapticLog, hasLength(1));
+      expect(
+        hapticLog.last,
+        isMethodCall('HapticFeedback.vibrate', arguments: 'HapticFeedbackType.mediumImpact'),
+      );
+
+      // Move the slider to the start slowly.
+      thumbCenter = tester.getTopRight(find.byKey(sliderKey)) - const Offset(unit, -unit);
+      await tester.timedDragFrom(
+        thumbCenter,
+        -Offset(thumbCenter.dx - topLeft.dx - unit * 2, 0),
+        const Duration(milliseconds: 1100),
+      );
+
+      expect(value, equals(0.0));
+
+      // Selection click should be emitted when the slider is slowly moved to the start.
+      expect(hapticLog, hasLength(2));
+      expect(
+        hapticLog.last,
+        isMethodCall('HapticFeedback.vibrate', arguments: 'HapticFeedbackType.selectionClick'),
+      );
+    },
+    variant: TargetPlatformVariant.only(TargetPlatform.iOS),
```

Comment:
Nit: Would it be valuable to run this test on all platforms and test that they do not emit haptic feedback? Probably overkill... I guess the vast majority of people will use this only on iOS.